Extendclaim enhancement

MaksyKun opened the pull request:
added permission for extendclaim to not require modification tool. In my case, people wished for expanding without the tool and I think its a benefit for the plugin too.

Dont know if `griefprevention.extendclaim` is appropriate to you, this is something that just came in my mind

MaksyKun committed:
fix for cauldron interactions on claims

MaksyKun committed:
added permission for extendclaim to not require modification tool

MaksyKun committed:
Merge branch 'master' into extendclaim-enhancement

Jikoo commented on src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java:
```diff
@@ -984,8 +984,8 @@ else if (GriefPrevention.instance.claimsEnabledForWorld(player.getLocation().get
-            //requires claim modification tool in hand
-            if (player.getGameMode() != GameMode.CREATIVE && player.getItemInHand().getType() != GriefPrevention.instance.config_claims_modificationTool)
+            // Unless the player has the extendclaim permission, he requires claim modification tool in hand
+            if (player.getGameMode() != GameMode.CREATIVE && (player.getItemInHand().getType() != GriefPrevention.instance.config_claims_modificationTool && !player.hasPermission("griefprevention.extendclaim")))
```
This is functional, but you don't need the parenthesis, and it might be more readable if broken out into separate lines.
```java
            if (player.getGameMode() != GameMode.CREATIVE
                    && player.getItemInHand().getType() != GriefPrevention.instance.config_claims_modificationTool
                    && !player.hasPermission("griefprevention.extendclaim"))
```
Honestly not sure if we have a convention for multi-line boolean logic. I prefer operator at the start because it's easier for me to follow, but I know there are areas in the code where it's at the end.

MaksyKun replied:
Id like to go with `griefprevention.extendclaim.no_tool` making permissions inheritable seems nice for future enhancements.  

Just updated the code. Could I ask what you mean exactly by putting the operator on the start? What I understand is like a reversed paranthesis to do 
```java
if(!(player.getGameMode() == GameMode.CREATIVE) 
    && player.getItemInHand().getType() == GriefPrevention.instance.config_claims_modificationTool
    && player.hasPermission("griefprevention.extendclaim")) {
    // Stuff to cancel
    }

```
But Im not sure if I got you right here

Jikoo replied:
Ah, no, I meant that elsewhere in the code there are multi-line logic expressions expressed
```java
            if (player.getGameMode() != GameMode.CREATIVE &&
                    player.getItemInHand().getType() != GriefPrevention.instance.config_claims_modificationTool &&
                    !player.hasPermission("griefprevention.extendclaim"))
```
but it's inconsistent and I'm not sure what our actual style choice is. Personally I prefer the way I wrote it initially.

MaksyKun replied:
Im not sure if there were any more changes required about this here. Its been a while, haha.
Could you recheck it and let me know?

Jikoo reviewed (approved):
I like the idea. I feel like `griefprevention.extendclaim` isn't very descriptive of what the node is intended to do - maybe `griefprevention.extendclaim.no_tool` or `griefprevention.config.extendclaim_no_tool` or something?

Giving this a general seal of approval because I don't know what direction Robo wants to take permission structure. Personally, I would like to see more detailed permissions added that can be inherited from the existing permissions as logical.

MaksyKun committed:
changed permission and removed parenthesis

MaksyKun committed:
Merge remote-tracking branch 'origin/extendclaim-enhancement' into extendclaim-enhancement

RoboMWM reviewed (changes requested):
The tool requirement was because bigscary designed GP to be something gradually introduced to players. I'm not sure how I want to address this requirement for players who already know how to claim and/or expect to do so on a server.

Anyways, for this PR, I'd change the wording from `no` to `bypass`. Still undecided on how it should be named though, the underscore is weird. Maybe `griefprevention.extendclaim.toolbypass`?

Also need to add to plugin.yml, and PR to documentation as well.

MaksyKun committed:
changed permission to toolbypass

MaksyKun committed:
added permission to plugin.yml

MaksyKun committed:
updated description

RoboMWM committed:
retain more of original comment

Not like there's consistency required for code comments, but I prefer to retain the original comment if possible and have at least some degree of relative consistency.

RoboMWM commented on src/main/resources/plugin.yml:
```diff
@@ -211,6 +211,9 @@ permissions:
             griefprevention.seeclaimsize: true
+    griefprevention.extendclaim.toolbypass:
+        description: Allows a player to extend a claim without a modification tool.
+        default: op
```
I'm thinking this should be default `false` to both maintain existing behavior and so admins don't have a different experience than players. (GP is intended to operate similarly for both players and admins, with /ignoreclaims being the "sudo" toggle if necessary.)

MaksyKun replied:
Got it changed

RoboMWM committed:
creative is also the exception

MaksyKun committed:
Made toolbypass default to false

RoboMWM committed:
Merge branch 'master' into extendclaim-enhancement
